### PR TITLE
Remove requirements that factors are created outside

### PR DESCRIPTION
**Pull request by anniejw6**
* Remove requirements that factors created outside
* Add more tests
* Minor cosmetic changes to functions

### CODE REVIEW

File: R/mod_marg.R [hunk 1]
```diff
@@ -18,18 +18,21 @@
...
+#' mtcars$gear <- as.character(mtcars$gear)
 #' mod <- glm(vs ~ gear + mpg * disp, data = mtcars, family = 'binomial')
 #' mod_marg2(mod, var_interest = 'gear',
 #'           type = 'levels', at = list(mpg = c(15, 21), disp = c(140, 180)))
 #'
 #' data(margex)
-#' margex$treatment <- factor(margex$treatment)
 #' mod <- glm(outcome ~ treatment + distance, data = margex, family = 'binomial')
 #' mod_marg2(mod, var_interest = 'treatment', type = 'levels', at = NULL)
 #' mod_marg2(mod, var_interest = 'treatment', type = 'effects', at = NULL)
```
Comment (nathanielolin) [inline R/mod_marg.R]: @anniejw6 I think these examples need to be updated. Treatment here is running as a continuous variable and the effects line fails with an error (which is working as intended I think, but probably not a great example?)

File: tests/testthat/test_overall.R [hunk 2]
```diff
@@ -1,37 +1,99 @@
 library(modmarg)
 context("Calculate everything correctly")
 
-test_that("output is calculated correctly", {
+test_that("levels are calculated correctly despite different input types", {
 
   # http://www.stata.com/support/faqs/statistics/compute-standard-errors-with-margins/
+
+  # Put as.factor in the equation
```
Comment (nathanielolin) [inline tests/testthat/test_overall.R]: This is as.character not as.factor

File: tests/testthat/test_overall.R [hunk 3]
```diff
@@ -1,37 +1,99 @@
...
+               c(.0655016, .2381135), tolerance = 0.0001)
+  expect_equal(z1$`Upper CI (95%)`, z2$`Upper CI (95%)`, z3$`Upper CI (95%)`,
+               c(.0927277, .2819272), tolerance = 0.0001)
+
+})
+
+
+test_that("effects are calculated correctly despite different input types", {
+
+  # Put as.factor in the equation
```
Comment (nathanielolin) [inline tests/testthat/test_overall.R]: as.character not as.factor

File: tests/testthat/test_overall.R [hunk 4]
```diff
@@ -1,37 +1,99 @@
...
+  expect_equal(z1$P.Value, z2$P.Value, z3$P.Value,
+               c(0), tolerance = 0.001)
+  expect_equal(z1$`Lower CI (95%)`, z2$`Lower CI (95%)`, z3$`Lower CI (95%)`,
+               c(.1550961), tolerance = 0.0001)
+  expect_equal(z1$`Upper CI (95%)`, z2$`Upper CI (95%)`, z3$`Upper CI (95%)`,
+               c(.2067153), tolerance = 0.0001)
+
+})
+
+test_that("works correctly even when terms are dropped", {
```
Comment (nathanielolin) [inline tests/testthat/test_overall.R]: Should be when *rows* are dropped; test for dropping covariates is separate.

File: R/mod_marg.R (hunk 1 above)
Comment (nathanielolin) [inline R/mod_marg.R]: I think these examples need to be updated. Treatment here is running as a continuous variable and the effects line fails with an error (which is working as intended I think, but probably not a great example?)

File: tests/testthat/test_overall.R (hunk 2 above)
Comment (nathanielolin) [inline tests/testthat/test_overall.R]: This is as.character not as.factor

File: tests/testthat/test_overall.R (hunk 3 above)
Comment (nathanielolin) [inline tests/testthat/test_overall.R]: as.character not as.factor

File: tests/testthat/test_overall.R (hunk 4 above)
Comment (nathanielolin) [inline tests/testthat/test_overall.R]: Should be when rows are dropped; test for dropping covariates is separate.

**Comment by anniejw6** [comment]
@nathanielolin - responded to your comments.

**Pull request merged**